src/smart-components/user/user.js

```diff
@@ -0,0 +1,199 @@
+
+    const createRows = (data) => {
+        return data ? data.reduce((acc, { uuid, name, groups_in = [], modified, accessCount }, i) => ([
+            ...acc, {
+                uuid,
+                cells: [{ title: name, props: { component: 'th', isOpen: false }},
+                { title: `${groups_in.length}`, props: { isOpen: expanded[uuid] === 1 }},
+                { title: accessCount, props: { isOpen: expanded[uuid] === 2 }},
+                { title: <DateFormat type='exact' date={ modified } /> }
+                ]
+            }, {
+                uuid: `${uuid}-groups`,
+                parent: 3 * i,
+                compoundParent: 1,
+                cells: [{
+                    props: { colSpan: 4, className: 'pf-m-no-padding' },
+                    title: <Table
+                        aria-label="Simple Table"
+                        variant={ TableVariant.compact }
+                        cells={ [ 'Name', 'Description' ] }
+                        rows={ groups_in.map(g => (
+                            { cells: [{ title: <Link to={ `/groups/detail/${g.uuid}` }>{g.name}</Link> }, g.description ]}
+                        )) }>
+                        <TableHeader />
+                        <TableBody />
+                    </Table>
+                }]
+            },
+            {
+                uuid: `${uuid}-groups`,
+                parent: 3 * i,
+                compoundParent: 2,
+                cells: [{
+                    props: { colSpan: 4, className: 'pf-m-no-padding' },
+                    title: rolesWithAccess && rolesWithAccess[uuid]
+                        ? <Table
+                            aria-label="Simple Table"
+                            variant={ TableVariant.compact }
+                            cells={ [ 'Application', 'Resource type', 'Operation' ] }
+                            rows={ rolesWithAccess[uuid].access.map(access => {
+                                const permissions = access.permission.split(':');
+                                return { cells: permissions };
+                            }) }>
+                            <TableHeader />
+                            <TableBody />
+                        </Table>
+                        : <ListLoader />
+
+                }]
+            }
+        ]), []) : [];
+    };
+
+    useEffect(() => {
+        fetchRoles({ limit: 20, offset: 0, addFields: [ 'groups_in' ], username });
+        debouncedFetch = debounce(
+                (limit, offset, name, addFields, username) => fetchRoles({ limit, offset, name, addFields, username }),
+            500);
+
+    }, []);
+
+    useEffect(() => {
+        if (!isLoading) {
+            setRows(createRows(roles.data));
+        }
+    }, [ roles ]);
+
+    useEffect(() => {
+        setRows(createRows(roles.data));
+    }, [ rolesWithAccess ]);
+
+    const onExpand = (event, rowIndex, colIndex, isOpen, rowData) => {
+        const r = [ ...rows ];
```

Comment:
I don't think you'd have to copy the rows in here. You can just use `rows`.

Reply:
Actually I don't need it at all :laughing: